plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1660,7 +1660,17 @@ public static void suspendAll() {
                     CpsFlowExecution cpsExec = (CpsFlowExecution) execution;
                     try {
                         cpsExec.checkAndAbortNonresumableBuild();
-
+                        if (cpsExec.owner != null) {
+                            try {
+                                Queue.Executable exec = cpsExec.owner.getExecutable();
+                                if (exec instanceof Saveable) {
+                                    LOGGER.fine(() -> "saving " + exec);
+                                    ((Saveable) exec).save();
+                                }
+                            } catch (IOException x) {
+                                LOGGER.log(Level.WARNING, "failed to save " + cpsExec, x);
+                            }
+                        }
```

Comment:
or perhaps just
```suggestion
                        cpsExec.saveOwner();
                            try {
                                Queue.Executable exec = cpsExec.owner.getExecutable();
                                if (exec instanceof Saveable) {
                                    LOGGER.fine(() -> "saving " + exec);
                                    ((Saveable) exec).save();
                                }
                            } catch (IOException x) {
                                LOGGER.log(Level.WARNING, "failed to save " + cpsExec, x);
                            }
                        }
```
though OTOH does https://github.com/jenkinsci/workflow-cps-plugin/blob/dc9d3aefcc6a28e8b5efa430a23d5b04ae0d92d1/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java#L1675 not already do just this?